drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -479,6 +474,68 @@ func getPortworxServiceSpec(
 	return newService
 }
 
+func (c *portworxBasic) createPortworxKVDBService(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	service := getPortworxKVDBServiceSpec(cluster, ownerRef)
+	return k8sutil.CreateOrUpdateService(c.k8sClient, service, ownerRef)
+}
+
+func getPortworxKVDBServiceSpec(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) *v1.Service {
+	labels := pxutil.SelectorLabels()
+	startPort := pxutil.StartPort(cluster)
+	kvdbTargetPort, _, _ := getTargetPorts(startPort)
+
+	newService := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pxutil.PortworxKVDBServiceName,
+			Namespace: cluster.Namespace,
+			Labels:    labels,
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{
+				constants.LabelKeyKVDBPod: constants.LabelValueTrue,
+			},
+			Type: v1.ServiceTypeClusterIP,
+			Ports: []v1.ServicePort{
+				{
+					Name:       pxutil.PortworxKVDBPortName,
+					Protocol:   v1.ProtocolTCP,
+					Port:       int32(9019),
+					TargetPort: intstr.FromInt(kvdbTargetPort),
+				},
+			},
+		},
+	}
+
+	if ownerRef != nil {
+		newService.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+	}
+
+	serviceType := pxutil.ServiceType(cluster)
+	if serviceType != "" {
+		newService.Spec.Type = serviceType
+	}
+
+	return newService
+}
+
+func getTargetPorts(startPort int) (int, int, int) {
+	kvdbTargetPort := 9019
+	sdkTargetPort := 9020
```

Comment:
Same question for restGetway.  Shouldn't it be `restGetwayDefaultPort` or `restGetwayServicePort`?

Reply:
Same as above.